cpp/Platform.Delegates/Platform.Delegates.MulticastDelegate.h

```diff
@@ -126,57 +88,98 @@ namespace Platform::Delegates
 
         MulticastDelegate &operator+=(const DelegateType &callback)
         {
-            const std::lock_guard<std::mutex> lock(mutex);
+            std::lock_guard lock(mutex);
             this->callbacks.emplace_back(callback);
             return *this;
         }
 
-        MulticastDelegate &operator+=(DelegateRawFunctionType &callback)
+        MulticastDelegate &operator+=(DelegateRawFunctionType callback)
         {
-            return *this += DelegateType(callback);
+            return *this += DelegateType{callback};
         }
 
         MulticastDelegate &operator+=(const DelegateFunctionType &callback)
         {
-            return *this += DelegateType(callback);
+            return *this += DelegateType{callback};
         }
 
         MulticastDelegate &operator-=(const DelegateType &callback)
         {
-            const std::lock_guard<std::mutex> lock(mutex);
+            const std::lock_guard lock{mutex};
```

Comment:
Wow! We can use type inference here using curly brackets?

Reply:
Type inference will work anyway, curly braces are just the preferred way to initialize objects in C++